connect/mirror/src/test/java/org/apache/kafka/connect/mirror/OffsetSyncStoreTest.java

```diff
@@ -119,7 +118,7 @@ public void testPastOffsetTranslation() {
                 store.sync(tp, offset, offset);
                 assertSparseSyncInvariant(store, tp);
```

Comment:
Can you move this into backingStoreStart, so that it happens when initializationMustReadToEnd is properly initialized? Here it's relying on the default value set by construction, not the value passed into `start`.

Actually this seems to be the case in a lot of the tests here. Can you look through the tests, and whenever there are assertions or `sync` calls before `start`, apply the same fix?